### PR TITLE
Fix Bug 1401464 - Appropriate label required on Edit Top Site dialog

### PR DESCRIPTION
**Pull request by piatra**
I've extracted the form input refactor from the custom screenshot patch into a separate PR.
This also includes some of the design from the custom screenshot PR
* preview for topsite when editing
* clear button for url input
* box shadow on focus
* some changes to size and padding

This will make the custom screenshot PR smaller and focused on just the screenshot part.

### CODE REVIEW

**Comment by Mardak** [comment]
@aaronrbenson This look right for "Add" in preparation for the custom screenshot?

image

**Comment by aaronrbenson** [comment]
Here's an updated spec with a title in the modal. I had to reorganize things a bit because the title looked out of place when there was a top site image on the left. Talked to @piatra about it and it shouldn't create too big a headache. 
  
screen shot 2018-02-12 at 10 39 43 am

**Comment by piatra** [comment]
Updated
screen shot 2018-02-12 at 19 37 24
screen shot 2018-02-12 at 19 37 10

File: system-addon/content-src/components/TopSites/_TopSites.scss [hunk 1]
```diff
@@ -1,15 +1,15 @@
...
-  $top-sites-border-radius: 6px;
-  $top-sites-title-height: 30px;
-  $top-sites-vertical-space: 8px;
-  $screenshot-size: cover;
-  $rich-icon-size: 96px;
-  $default-icon-wrapper-size: 42px;
-  $default-icon-size: 32px;
-  $default-icon-offset: 6px;
-  $half-base-gutter: $base-gutter / 2;
+$top-sites-size: $grid-unit;
```
Comment (Mardak) [inline system-addon/content-src/components/TopSites/_TopSites.scss]: This type of change is one of the main reason to keep various refactoring to their own commit. These indentation changes makes it incredibly difficult to review functionality changes when there's a ton of unrelated changes mixed in.

Comment (piatra) [reply system-addon/content-src/components/TopSites/_TopSites.scss]: Sorry about that, I can take this into it's own commit.

Comment (Mardak) [reply system-addon/content-src/components/TopSites/_TopSites.scss]: I think I've worked through most of it manually comparing with two windows for side-by-side to figure out what was just indentation / moving up/down vs what's changes.

**Comment by piatra** [comment]
@Mardak I've tried to re-commit the changes. Let me know if this help/how to improve for future PRs.

File: system-addon/content-src/components/TopSites/TopSiteForm.jsx [hunk 2]
```diff
@@ -102,33 +98,28 @@ export class TopSiteForm extends React.PureComponent {
...
+              value={this.state.url}
+              onClear={this.onClearUrlClick}
+              validationError={this.state.validationError}
+              titleId="topsites_form_url_label"
+              placeholderId="topsites_form_url_placeholder"
+              errorMessageId="topsites_form_url_validation"
+              {...this.props} />
           </div>
-        </section>
+          <TopSiteLink link={this.props.site || {}} title={this.state.label} preventNavigation={true} />
```
Comment (Mardak) [inline system-addon/content-src/components/TopSites/TopSiteForm.jsx]: Why is this inside the container div instead of the form-wrapper? In particular, is it expected that the `<h3>` title wraps at the same width as the form inputs? E.g.,
```
Title title
title
[ input    ] [link]
```

vs
```
Title title title
[ input    ] [link]
```

File: system-addon/content-src/components/TopSites/TopSiteForm.jsx [hunk 3]
```diff
@@ -102,33 +98,28 @@ export class TopSiteForm extends React.PureComponent {
...
-                <aside className="error-tooltip">
-                  <FormattedMessage id="topsites_form_url_validation" />
-                </aside>
-              }
-            </div>
+            <TopSiteFormInput onChange={this.onLabelChange}
+              value={this.state.label}
+              titleId="topsites_form_title_label"
+              placeholderId="topsites_form_title_placeholder"
+              errorMessageId="topsites_form_url_validation"
```
Comment (Mardak) [inline system-addon/content-src/components/TopSites/TopSiteForm.jsx]: The title won't be showing an error message.

File: system-addon/content-src/components/TopSites/TopSiteForm.jsx [hunk 4]
```diff
@@ -102,33 +98,28 @@ export class TopSiteForm extends React.PureComponent {
...
+              {...this.props} />
+            <TopSiteFormInput onChange={this.onUrlChange}
+              value={this.state.url}
+              onClear={this.onClearUrlClick}
+              validationError={this.state.validationError}
+              titleId="topsites_form_url_label"
+              placeholderId="topsites_form_url_placeholder"
+              errorMessageId="topsites_form_url_validation"
+              {...this.props} />
```
Comment (Mardak) [inline system-addon/content-src/components/TopSites/TopSiteForm.jsx]: What props are we expecting to pass on to the input?

File: system-addon/content-src/components/TopSites/TopSiteForm.jsx (hunk 2 above)
Comment (Mardak) [reply system-addon/content-src/components/TopSites/TopSiteForm.jsx]: (Maybe `h3` should be moved out like link?)

File: system-addon/content-src/components/TopSites/_TopSites.scss [hunk 5]
```diff
@@ -326,13 +321,54 @@
 }
 
 .topsite-form {
+  .form-input-container {
+    display: flex;
+    max-width: 492px;
```
Comment (Mardak) [inline system-addon/content-src/components/TopSites/_TopSites.scss]: What are we expecting to grow here to want this particular max-width? If it's the appropriate value, probably pull it out to a variable for the `.topsite-form`.

File: system-addon/content-src/components/TopSites/_TopSites.scss [hunk 6]
```diff
@@ -326,13 +321,54 @@
...
+    margin: 0 auto;
+    padding: 32px;
+
+    .top-site-outer {
+      width: 96px;
```
Comment (Mardak) [inline system-addon/content-src/components/TopSites/_TopSites.scss]: Is this width necessary? Seems like it's something a regular top site link would have / want?

File: system-addon/content-src/components/TopSites/_TopSites.scss [hunk 7]
```diff
@@ -326,13 +321,54 @@
...
+      padding: 0;
+      margin: 64px 0 0;
+      margin-inline-start: 32px;
+      list-style: none;
```
Comment (Mardak) [inline system-addon/content-src/components/TopSites/_TopSites.scss]: Seems odd that we need to specially do this here for the form. Looks like the list normally avoids this by `display: inline-block;`.. maybe it should just happen for all `.top-site-outer`?

File: system-addon/content-src/components/TopSites/_TopSites.scss [hunk 8]
```diff
@@ -326,13 +321,54 @@
...
+
+  label {
+    font-size: 13px;
+  }
+
   .form-wrapper {
-    margin: auto;
-    max-width: 350px;
-    padding: 15px 0;
+    width: 300px;
```
Comment (Mardak) [inline system-addon/content-src/components/TopSites/_TopSites.scss]: What's this change from max to a smaller fixed width? Are we not expecting some stretching anymore?

File: system-addon/content-src/components/TopSites/_TopSites.scss [hunk 9]
```diff
@@ -326,13 +321,54 @@
...
+        width: 24px;
+        height: 16px;
+        background: $white url('#{$image-path}searchfield-cancel.svg');
+        cursor: pointer;
+        background-size: 16px;
+        background-repeat: no-repeat;
+        background-position: center;
+
+        &:dir(rtl) {
+          left: 4px;
```
Comment (Mardak) [inline system-addon/content-src/components/TopSites/_TopSites.scss]: These can be done with `offset-inline-end`

File: system-addon/content-src/components/TopSites/_TopSites.scss [hunk 10]
```diff
@@ -326,13 +321,54 @@
...
       position: relative;
+
+      .clear-input-value {
+        position: absolute;
+        top: 16px;
+        width: 24px;
+        height: 16px;
+        background: $white url('#{$image-path}searchfield-cancel.svg');
+        cursor: pointer;
+        background-size: 16px;
```
Comment (Mardak) [inline system-addon/content-src/components/TopSites/_TopSites.scss]: Just making sure, the svg file is sized for 14 but used as 16 here?

File: system-addon/content-src/components/TopSites/_TopSites.scss [hunk 11]
```diff
@@ -326,13 +321,54 @@
...
-    padding: 15px 0;
+    width: 300px;
 
     .field {
       position: relative;
+
+      .clear-input-value {
+        position: absolute;
+        top: 16px;
+        width: 24px;
```
Comment (Mardak) [inline system-addon/content-src/components/TopSites/_TopSites.scss]: Is the click target intended to be wider than the image?

File: system-addon/content-src/components/TopSites/_TopSites.scss [hunk 12]
```diff
@@ -326,13 +321,54 @@
...
+        height: 16px;
+        background: $white url('#{$image-path}searchfield-cancel.svg');
```
Comment (Mardak) [inline system-addon/content-src/components/TopSites/_TopSites.scss]: Does this explicit background color need to be set? Might cause oddness if the input gets a different color somehow. If not, probably just set `background-image` ?

File: system-addon/content-src/components/TopSites/_TopSites.scss [hunk 13]
```diff
@@ -341,19 +377,26 @@
     }
 
     .section-title {
-      margin-bottom: 5px;
+      text-transform: none;
```
Comment (Mardak) [inline system-addon/content-src/components/TopSites/_TopSites.scss]: Probably good to double check if this was an intended design change to avoid the upper casing that is used for all other section titles requiring overriding here.

File: system-addon/content-src/components/TopSites/_TopSites.scss [hunk 14]
```diff
@@ -341,19 +377,26 @@
...
+      font-size: 16px;
+      font-weight: bold;
```
Comment (Mardak) [inline system-addon/content-src/components/TopSites/_TopSites.scss]: It should already be bold.

File: system-addon/content-src/components/TopSites/_TopSites.scss [hunk 15]
```diff
@@ -326,13 +321,54 @@
...
-    max-width: 350px;
-    padding: 15px 0;
+    width: 300px;
 
     .field {
       position: relative;
+
+      .clear-input-value {
+        position: absolute;
+        top: 16px;
```
Comment (Mardak) [inline system-addon/content-src/components/TopSites/_TopSites.scss]: This css seems odd, but maybe it just needs to share some whitespace variable with the input?

File: system-addon/content-src/components/TopSites/_TopSites.scss [hunk 16]
```diff
@@ -326,13 +321,54 @@
 }
 
 .topsite-form {
+  .form-input-container {
+    display: flex;
```
Comment (Mardak) [inline system-addon/content-src/components/TopSites/_TopSites.scss]: What needs to flex here? I believe oddness happens on narrow when things try to overflow with various other widths specified elsewhere.

File: system-addon/content-src/components/TopSites/_TopSites.scss [hunk 17]
```diff
@@ -326,13 +321,54 @@
...
+    max-width: 492px;
+    margin: 0 auto;
+    padding: 32px;
+
+    .top-site-outer {
+      width: 96px;
+      padding: 0;
+      margin: 64px 0 0;
+      margin-inline-start: 32px;
```
Comment (Mardak) [inline system-addon/content-src/components/TopSites/_TopSites.scss]: Thought I had a comment here but seems like it didn't make it into the review…? This particular positioning seems odd. Is it expected to line up with the top of the first input box? What happens when there's more lines or more inputs?

screen shot 2018-02-12 at 3 20 25 pm

File: system-addon/content-src/components/TopSites/_TopSites.scss (hunk 13 above)
Comment (piatra) [reply system-addon/content-src/components/TopSites/_TopSites.scss]: Yes, @aaronrbenson 
> the all-caps version just doesn’t look good in this case
> trying to use all-caps very sparingly

File: system-addon/content-src/components/TopSites/_TopSites.scss (hunk 12 above)
Comment (piatra) [reply system-addon/content-src/components/TopSites/_TopSites.scss]: There is actually no design for this. I was trying to avoid input overflow scenarios where it looks odd with no background color. I will ask for design feedback.
screen shot 2018-02-13 at 09 51 49183
screen shot 2018-02-13 at 09 52 04107

Comment (piatra) [reply system-addon/content-src/components/TopSites/_TopSites.scss]: > Yep. Some padding is preferred

File: system-addon/content-src/components/TopSites/TopSiteForm.jsx [hunk 18]
```diff
@@ -15,18 +17,23 @@ export class TopSiteForm extends React.PureComponent {
...
-    this.resetValidation();
-    this.setState({"url": event.target.value});
+    this.setState({
+      url: event.target.value,
+      validationError: false
+    });
+  }
+
+  onClearUrlClick() {
+    this.setState({url: ""});
```
[the remark this answers is not in the document]
Comment (piatra) [reply system-addon/content-src/components/TopSites/TopSiteForm.jsx]: I did not intend for this but while testing I realized it *could* make sense: an empty url is not valid. if you clear and save you just get the error again.

[the remark this answers is not in the document]
Comment (piatra) [reply system-addon/content-src/components/TopSites/TopSiteForm.jsx]: But probably not what the users expects.

File: system-addon/content-src/components/TopSites/_TopSites.scss (hunk 17 above)
Comment (piatra) [reply system-addon/content-src/components/TopSites/_TopSites.scss]: To fix this I added a media query so that on a narrow design the preview moves below the inputs.
This seems good enough for our current fields. Even if we add more fields with longer labels (Custom Screenshot URL might be one culprit) it won't break alignment since only the first input matters.
Design feedback:
> does this look ok for a narrow view?
> Yeah that’s fine

screen shot 2018-02-13 at 17 11 31

**Comment by Mardak** [comment]
There are many review comments that are still unanswered. At least respond to each of them.

File: system-addon/content-src/components/TopSites/TopSiteFormInput.jsx [hunk 19]
```diff
@@ -0,0 +1,45 @@
...
+
+export class TopSiteFormInput extends React.PureComponent {
+  constructor(props) {
+    super(props);
+    this.onMount = this.onMount.bind(this);
+  }
+
+  componentWillReceiveProps(nextProps) {
+    if (nextProps.validationError) {
+      this.input.focus();
```
[the remark this answers is not in the document]
Comment (piatra) [reply system-addon/content-src/components/TopSites/TopSiteFormInput.jsx]: Agreed. But is this something worth fixing when I'll land the custom screenshot input? If both fields have errors both need to be fixed, where focus lands seems less important.

File: system-addon/content-src/components/TopSites/TopSiteForm.jsx (hunk 2 above)
Comment (piatra) [reply system-addon/content-src/components/TopSites/TopSiteForm.jsx]: I ended up wrapping the inputs and preview together in a div to be it easier to reason about the position of the preview in relation to the first input field. It should be (16px for the text + 8px margin) from the top.
And move the title outside so that it can match your 2nd example.

File: system-addon/content-src/components/TopSites/TopSiteForm.jsx (hunk 4 above)
Comment (piatra) [reply system-addon/content-src/components/TopSites/TopSiteForm.jsx]: The only other prop needed was `intl`

File: system-addon/content-src/components/TopSites/_TopSites.scss (hunk 6 above)
Comment (piatra) [reply system-addon/content-src/components/TopSites/_TopSites.scss]: I ended up removing this but the problem was caused by the `<a>` having display block.

File: system-addon/content-src/components/TopSites/_TopSites.scss (hunk 5 above)
Comment (piatra) [reply system-addon/content-src/components/TopSites/_TopSites.scss]: I created 
```
$form-width: 300px;
$form-spacing: 32px;
```
And defined the width of the input + preview using those variables. The `.form-wrapper` has width 100% and can grow or shrink (the input width) when browser size is between the breakpoint (defined for the form) and full width.

File: system-addon/content-src/components/TopSites/_TopSites.scss (hunk 7 above)
Comment (piatra) [reply system-addon/content-src/components/TopSites/_TopSites.scss]: Updated so that all `.top-site-outer` have display inline.

File: system-addon/content-src/components/TopSites/_TopSites.scss (hunk 10 above)
Comment (piatra) [reply system-addon/content-src/components/TopSites/_TopSites.scss]: Updated to a 16x svg.

File: system-addon/content-src/components/TopSites/_TopSites.scss (hunk 11 above)
Comment (piatra) [reply system-addon/content-src/components/TopSites/_TopSites.scss]: No. I changed the markup to have a white background container for the image. The container has `pointer-events: none` so that only clicking the image will clear the input value.

File: system-addon/content-src/components/TopSites/_TopSites.scss (hunk 15 above)
Comment (piatra) [reply system-addon/content-src/components/TopSites/_TopSites.scss]: @Mardak I'm not sure what the fix would be here.

File: system-addon/content-src/components/TopSites/_TopSites.scss (hunk 16 above)
Comment (piatra) [reply system-addon/content-src/components/TopSites/_TopSites.scss]: The `display: flex` declaration moved to the container I mentioned [in this comment](https://github.com/mozilla/activity-stream/pull/3965#discussion_r168810824) that wraps input and preview together. I used it to place both elements as siblings while allowing the preview to have a margin top to align with the input and not shift its sibling.
Do you still think this need to change considering the latest way it's implemented?

File: system-addon/data/content/assets/searchfield-cancel.svg [hunk 20]
```diff
@@ -0,0 +1,20 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<!-- This Source Code Form is subject to the terms of the Mozilla Public
+   - License, v. 2.0. If a copy of the MPL was not distributed with this
+   - file, You can obtain one at http://mozilla.org/MPL/2.0/. -->
+<svg xmlns="http://www.w3.org/2000/svg" xmlns:xlink="http://www.w3.org/1999/xlink" width="14" height="14" viewBox="0 0 14 14">
```
[the remark this answers is not in the document]
Comment (piatra) [reply system-addon/data/content/assets/searchfield-cancel.svg]: I changed the asset.

File: system-addon/content-src/components/TopSites/TopSite.jsx [hunk 21]
```diff
@@ -101,7 +101,7 @@ export class TopSiteLink extends React.PureComponent {
     }
     return (<li className={topSiteOuterClassName} onDrop={this.onDragEvent} onDragOver={this.onDragEvent} onDragEnter={this.onDragEvent} onDragLeave={this.onDragEvent} {...draggableProps}>
       <div className="top-site-inner">
-         <a href={link.url} onClick={onClick}>
+         <a href={!this.props.preventNavigation ? link.url : ""} onClick={onClick}>
```
[the remark this answers is not in the document]
Comment (piatra) [reply system-addon/content-src/components/TopSites/TopSite.jsx]: I used this suggestion and reverted the change.

File: system-addon/data/content/assets/searchfield-cancel.svg [hunk 22]
```diff
@@ -0,0 +1,20 @@
...
+  <style>
+    circle {
+      fill: #808080;
```
[the remark this answers is not in the document]
Comment (piatra) [reply system-addon/data/content/assets/searchfield-cancel.svg]: Used a different asset and used fill to color it.

File: system-addon/content-src/components/TopSites/_TopSites.scss (hunk 8 above)
Comment (piatra) [reply system-addon/content-src/components/TopSites/_TopSites.scss]: Stretching is now done by the form-wrapper as I [describe here](https://github.com/mozilla/activity-stream/pull/3965#discussion_r168813353)

File: system-addon/content-src/components/TopSites/_TopSites.scss [hunk 23]
```diff
@@ -326,34 +320,89 @@
...
+
+      .clear-input-value-container {
+        position: absolute;
+        top: 16px;
+        width: 24px;
+        background: $white;
+        pointer-events: none;
+
+        &:dir(rtl) {
+          offset-inline-start: 1px;
```
Comment (Mardak) [inline system-addon/content-src/components/TopSites/_TopSites.scss]: This probably isn't what you wanted:
screen shot 2018-02-16 at 3 00 46 pm

Just have an `offset-inline-end: 1px` without a `:dir` check. It'll be as if it's `left: 1px` for `rtl` (and `right: 1px` for `ltr`).

File: system-addon/content-src/components/TopSites/_TopSites.scss [hunk 24]
```diff
@@ -326,34 +320,89 @@
...
+        }
+
+        .clear-input-value {
+          width: $icon-size;
+          height: $icon-size;
+          margin: 0 auto;
+          fill: $fill-secondary;
+          -moz-context-properties: fill;
+          background: url('#{$image-path}glyph-cancel-16.svg');
+          pointer-events: auto;
```
Comment (Mardak) [inline system-addon/content-src/components/TopSites/_TopSites.scss]: `auto` doesn't seem right. Our other buttons have `pointer`

File: system-addon/content-src/components/TopSites/_TopSites.scss [hunk 25]
```diff
@@ -326,34 +320,89 @@
...
+    width: 100%;
 
     .field {
       position: relative;
+
+      .clear-input-value-container {
+        position: absolute;
+        top: 16px;
+        width: 24px;
+        background: $white;
```
Comment (Mardak) [inline system-addon/content-src/components/TopSites/_TopSites.scss]: Covering up with a white background is probably incorrectly assuming what color the input box is. We can avoid the text overlap by having a `-moz-padding-end` on the input.

File: system-addon/content-src/components/TopSites/_TopSites.scss [hunk 26]
```diff
@@ -326,34 +320,89 @@
...
+    max-width: $form-width + 3 * $form-spacing + $rich-icon-size;
+    margin: 0 auto;
+    padding: $form-spacing $form-spacing 40px;
+
+    .top-site-outer {
+      padding: 0;
+      margin: 24px 0 0;
+      margin-inline-start: $form-spacing;
+      pointer-events: none;
+      cursor: default;
```
Comment (Mardak) [inline system-addon/content-src/components/TopSites/_TopSites.scss]: No need for this `cursor` as the pointer events are as if it's point at the the empty space behind it.

File: system-addon/content-src/components/TopSites/_TopSites.scss [hunk 27]
```diff
@@ -326,34 +320,89 @@
...
+    }
+
+    .section-title {
+      text-transform: none;
+      font-size: 16px;
+      line-height: 19px;
+      color: $grey-60;
```
Comment (Mardak) [inline system-addon/content-src/components/TopSites/_TopSites.scss]: This `color` isn't actually being used as the span has a color set.

File: system-addon/content-src/components/TopSites/_TopSites.scss [hunk 28]
```diff
@@ -326,34 +320,89 @@
...
+      margin: 24px 0 0;
+      margin-inline-start: $form-spacing;
+      pointer-events: none;
+      cursor: default;
+    }
+
+    .section-title {
+      text-transform: none;
+      font-size: 16px;
+      line-height: 19px;
```
Comment (Mardak) [inline system-addon/content-src/components/TopSites/_TopSites.scss]: This `line-height` doesn't seem to change anything with the `font-size: 16px`?

File: system-addon/content-src/components/TopSites/_TopSites.scss [hunk 29]
```diff
@@ -326,34 +320,89 @@
...
-    max-width: 350px;
-    padding: 15px 0;
+    width: 100%;
 
     .field {
       position: relative;
+
+      .clear-input-value-container {
+        position: absolute;
+        top: 16px;
```
Comment (Mardak) [inline system-addon/content-src/components/TopSites/_TopSites.scss]: I think the oddness is that this explicit `16px` value actually depends on the sizing of the input box. Maybe just vertically center it with:
```css
{
  transform: translateY(-50%);
  top: 50%;
}
```

File: system-addon/content-src/components/TopSites/TopSiteFormInput.jsx [hunk 30]
```diff
@@ -0,0 +1,46 @@
...
+  onMount(input) {
+    this.input = input;
+  }
+
+  render() {
+    const showClearButton = this.props.value && this.props.onClear;
+
+    return (<label><FormattedMessage id={this.props.titleId} />
+      <div className={`field url${this.props.validationError ? " invalid" : ""}`}>
+        {showClearButton && <div className="clear-input-value-container">
```
Comment (Mardak) [inline system-addon/content-src/components/TopSites/TopSiteFormInput.jsx]: Do we actually need this container now instead of just the "icon/button"?

File: system-addon/content-src/components/TopSites/_TopSites.scss (hunk 23 above)
Comment (Mardak) [reply system-addon/content-src/components/TopSites/_TopSites.scss]: You can test by opening up the inspector and changing the top level `<html dir="rtl">`

File: system-addon/data/content/assets/glyph-cancel-16.svg [hunk 31]
```diff
@@ -0,0 +1,6 @@
+<!-- This Source Code Form is subject to the terms of the Mozilla Public
+   - License, v. 2.0. If a copy of the MPL was not distributed with this
+   - file, You can obtain one at http://mozilla.org/MPL/2.0/. -->
+<svg xmlns="http://www.w3.org/2000/svg" width="16" height="16" viewBox="0 0 16 16">
+  <path fill="context-fill" fill-rule="evenodd" d="M6.586 8l-2.293 2.293a1 1 0 0 0 1.414 1.414L8 9.414l2.293 2.293a1 1 0 0 0 1.414-1.414L9.414 8l2.293-2.293a1 1 0 1 0-1.414-1.414L8 6.586 5.707 4.293a1 1 0 0 0-1.414 1.414L6.586 8zM8 0a8 8 0 1 1 0 16A8 8 0 0 1 8 0z"></path>
+</svg>
```
[the remark this answers is not in the document]
Comment (piatra) [reply system-addon/data/content/assets/glyph-cancel-16.svg]: This was taken directly from our website https://design.firefox.com/icons/viewer/ 
I'll request a different one from design.

File: system-addon/content-src/components/TopSites/_TopSites.scss (hunk 24 above)
Comment (piatra) [reply system-addon/content-src/components/TopSites/_TopSites.scss]: The one in `about:preferences` doesn't have a pointer.

File: system-addon/content-src/components/TopSites/_TopSites.scss (hunk 28 above)
Comment (piatra) [reply system-addon/content-src/components/TopSites/_TopSites.scss]: It was supposed to share the style of https://github.com/mozilla/activity-stream/blob/a050adee0eebcc47c310ed9d9fb9973f1a07e1bd/system-addon/content-src/components/PreferencesPane/_PreferencesPane.scss#L68-L71

File: system-addon/content-src/components/TopSites/_TopSites.scss (hunk 23 above)
Comment (piatra) [reply system-addon/content-src/components/TopSites/_TopSites.scss]: I did know about the `.rtl` class but I never actually typed something while in rtl mode and it looked ok to me..

File: system-addon/content-src/components/TopSites/_TopSites.scss [hunk 32]
```diff
@@ -61,177 +61,176 @@
...
-  &:not(.dnd-active) {
-    .top-site-outer:-moz-any(.active, :focus, :hover) {
-      .tile {
-        @include fade-in;
+  &.dragged {
+    .tile {
+      background: $grey-20;
+      box-shadow: none;
+
+      * {
```
[the remark this answers is not in the document]
Comment (piatra) [reply system-addon/content-src/components/TopSites/_TopSites.scss]: Yeah looks like `*` and `*, before` landed in separate commits and my rebase lost it.

File: system-addon/content-src/components/TopSites/_TopSites.scss [hunk 33]
```diff
@@ -326,34 +321,75 @@
...
       position: relative;
+
+      .icon-clear-input {
+        position: absolute;
+        transform: translateY(-50%);
+        top: 50%;
+        offset-inline-end: 8px;
+
+        & + input:dir(ltr) {
+          padding-inline-end: 28px;
```
Comment (Mardak) [inline system-addon/content-src/components/TopSites/_TopSites.scss]: Your comments and changes here made me realize this is a bit more complicate than I thought. For ltr, the padding should always be on the right, and for rtl, the padding should be on the left. This means we want padding-right and padding-left respectively because padding-inline-end will depend on the text direction of the input instead of on the document. This is particularly confusing because urls in rtl are specially treated to have text flow ltr but right aligned.

File: system-addon/content-src/components/TopSites/TopSiteFormInput.jsx [hunk 34]
```diff
@@ -0,0 +1,45 @@
...
+
+  onMount(input) {
+    this.input = input;
+  }
+
+  render() {
+    const showClearButton = this.props.value && this.props.onClear;
+
+    return (<label><FormattedMessage id={this.props.titleId} />
+      <div className={`field url${this.props.validationError ? " invalid" : ""}`}>
```
Comment (Mardak) [inline system-addon/content-src/components/TopSites/TopSiteFormInput.jsx]: This can't be `.url` for all inputs. Only for url type inputs, so not the title.

**Comment by as-pine-proxy** [comment]
[Treeherder: fd83d3e6307fef66435510d05765a3cc336c2277](https://treeherder.mozilla.org/#/jobs?repo=pine&revision=fd83d3e6307fef66435510d05765a3cc336c2277)
